File: .py/62.UniquePaths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        N = n + m - 2
        r = m - 1 if m<=n else n - 1
        result = 1
        for i in range(1,r+1):
            result = result * ((N - r + i)/ i )
        return round(result)

File: .cpp/62.UniquePaths.cpp
class Solution
{
public:
    int uniquePaths(int m, int n)
    {
        int N = m + n - 2;
        int r = m <= n ? m - 1 : n - 1;
        long long result = 1;
        for (int i = 1; i <= r; i++)
        {
            result = result * (N - r + i) / i;
        }
        return result;
    }
};